docs-source/content/samples/simple/elastic-stack/soa-domain/weblogic-monitoring-exporter-setup.md

```diff
@@ -0,0 +1,305 @@
+---
+title: "Monitor a SOA domain"
+date: 2019-12-05T06:46:23-05:00
+weight: 5
+description: "Use the WebLogic Monitoring Exporter to monitor a SOA instance using Prometheus and Grafana."
+---
+
+Monitor a SOA domain using Prometheus and Grafana by exporting the metrics from the domain instance using the
+WebLogic Monitoring Exporter. This sample shows you how to set up the WebLogic Monitoring Exporter to push the data
+to Prometheus.
+
+#### Prerequisites
+
+This document assumes that the Prometheus Operator is deployed on the Kubernetes cluster. If it is not already deployed, follow the steps below for deploying the Prometheus Operator.
+
+#### Clone the kube-prometheus project
+
+```bash
+$ git clone https://github.com/coreos/kube-prometheus.git
+```
+
+#### Create the kube-prometheus resources
+
+Change to the `kube-prometheus` directory and execute the following commands to create the namespace and CRDs. Wait for their availability before creating the remaining resources.
+
+```bash
+$ cd kube-prometheus
+
+$ kubectl create -f manifests/setup
+$ until kubectl get servicemonitors --all-namespaces ; do date; sleep 1; echo ""; done
+$ kubectl create -f manifests/
+```
+
+#### Label the nodes
+Kube-Prometheus requires all the exporter nodes to be labelled with `kubernetes.io/os=linux`. If a node is not labelled, then you must label it using the following command:
+
+```
+$ kubectl label nodes --all kubernetes.io/os=linux
+```
+#### Provide external access
+To provide external access for Grafana, Prometheus, and Alertmanager, execute the commands below:
+
+```bash
+$ kubectl patch svc grafana -n monitoring --type=json -p '[{"op": "replace", "path": "/spec/type", "value": "NodePort" },{"op": "replace", "path": "/spec/ports/0/nodePort", "value": 32100 }]'
+$ kubectl patch svc prometheus-k8s -n monitoring --type=json -p '[{"op": "replace", "path": "/spec/type", "value": "NodePort" },{"op": "replace", "path": "/spec/ports/0/nodePort", "value": 32101 }]'
+$ kubectl patch svc alertmanager-main -n monitoring --type=json -p '[{"op": "replace", "path": "/spec/type", "value": "NodePort" },{"op": "replace", "path": "/spec/ports/0/nodePort", "value": 32102 }]'
+```
+
+**NOTE**:
+
+* 32100 is the external port for Grafana
+* 32101 is the external port for Prometheus
+* 32102 is the external port for Alertmanager
+
+--------------
+
+Use the following instructions to set up the WebLogic Monitoring Exporter to collect WebLogic Server metrics and monitor a SOA domain.
+
+#### Download the WebLogic Monitoring Exporter
+
+Download these WebLogic Monitoring Exporter files from the [Releases](https://github.com/oracle/weblogic-monitoring-exporter/releases) page:
+
+* `wls-exporter.war`
+* `getX.X.X.sh`
+
+#### Create a configuration file for the WebLogic Monitoring Exporter
+
+The configuration file will have the server port for the web application metrics to be exported from the WebLogic Server.
```

Comment:
Please rephrase this statement as below:

The configuration file will have the server port of the WebLogic Server where the monitoring exporter application is deployed.

Reply:
Done.